src/zenml/integrations/github/orchestrators/github_actions_orchestrator.py

```diff
@@ -0,0 +1,450 @@
+            untracked_files=True
+        ):
+            raise RuntimeError(
+                "Trying to run a pipeline from within a dirty git repository."
+                "If you want this orchestrator to skip the dirty repo check in "
+                f"the future, run\n `zenml orchestrator update {self.name} "
+                "--prevent_dirty_repository=false`"
+            )
+
+        image_name = self.get_docker_image_name(pipeline.name)
+        requirements = {*stack.requirements(), *pipeline.requirements}
+
+        logger.debug(
+            "Github actions docker image requirements: %s", requirements
+        )
+
+        docker_utils.build_docker_image(
+            build_context_path=source_utils.get_source_root_path(),
+            image_name=image_name,
+            dockerignore_path=pipeline.dockerignore_file,
+            requirements=requirements,
+            base_image=self.custom_docker_base_image_name,
+        )
+
+        assert stack.container_registry  # should never happen due to validation
+        stack.container_registry.push_image(image_name)
+
+        # Store the docker image digest in the runtime configuration so it gets
+        # tracked in the ZenStore
+        image_digest = docker_utils.get_image_digest(image_name) or image_name
+        runtime_configuration["docker_image"] = image_digest
+
+    def prepare_or_run_pipeline(
+        self,
+        sorted_steps: List[BaseStep],
+        pipeline: "BasePipeline",
+        pb2_pipeline: Pb2Pipeline,
+        stack: "Stack",
+        runtime_configuration: "RuntimeConfiguration",
+    ) -> Any:
+        """Writes a GitHub Action workflow yaml and optionally pushes it."""
+        workflow_path = os.path.join(
+            self.workflow_directory, f"{pipeline.name}.yaml"
+        )
+
+        workflow_dict: Dict[str, Any] = {"name": pipeline.name}
+
+        schedule = runtime_configuration.schedule
+        if schedule:
+            if not schedule.cron_expression:
+                raise ValueError(
+                    "GitHub Action workflows can only be scheduled using cron "
+                    "expressions and not using a periodic schedule. If you "
+                    "want to schedule pipelines using this GitHub Action "
+                    "orchestrator, please include a cron expression in your "
+                    "schedule object."
+                )
+
+            # GitHub workflows requires a schedule interval of at least 5
+            # minutes. Invalid cron expressions would be something like
+            # `*/3 * * * *` (all stars except the first part of the expression,
+            # which will have the format `*/minute_interval`)
+            if re.fullmatch(r"\*/[1-4]( \*){4,}", schedule.cron_expression):
+                raise ValueError(
+                    "GitHub workflows requires a schedule interval of at "
+                    "least 5 minutes which is incompatible with your cron "
```

Comment:
Maybe add an example of a valid expression?

Reply:
Good idea  👍